### PR TITLE
Fix incorrect `_document.js` error when `disableStaticImages: true`

### LINKED ISSUE #30600: next 12: False-Positive Build Error: "Images cannot be imported within pages/_document.js"
### What version of Next.js are you using?

12.0.1 and 12.0.2-canary.8

### What version of Node.js are you using?

14.17.6

### What browser are you using?

none

### What operating system are you using?

macOS

### How are you deploying your application?

none

### Describe the Bug

During build, this error shows up:

```
Images cannot be imported within pages/_document.js. Please move image imports that need to be displayed on every page into pages/_app.js.
Read more: https://nextjs.org/docs/messages/custom-document-image-import
Location: pages/_document.tsx
```

Note that we **do not** use `next/image`. We use our own webpack plugin based on `file-loader` to import images and copy them to the build output. It works fine in next.js 11 with webpack 5. We even have set `images.disableStaticImages:true` in next.config.js (even though that should not be necessary since we later might use next/image for some images except the ones in _document.tsx).

### Expected Behavior

It should compile fine.

### To Reproduce

* Add an image import to _document.tsx: `import favicon16 from 'static_hashed/img/touch-icons/favicon-16x16.png'`
* Do not use next/image
* Run `next build` or `next dev`

Note that we use the imported images for `<Head><link rel="icon">`:

Our webpack plugin should not be necessary to demonstrate the issue, but let us know if you need it.

### PR DESCRIPTION
**Pull request by styfle**
Fixes #30600

### CODE REVIEW

File: errors/custom-document-image-import.md [hunk 1]
```diff
@@ -28,8 +28,19 @@ function MyApp({ Component, pageProps }) {
 export default MyApp
 ```
 
+If your application is not using `next/image`, you can disable the built-in loader with the following next.config.js:
```
Comment (ijjk) [inline errors/custom-document-image-import.md:31]:
```suggestion
If your application is not using image imports with `next/image`, you can disable the built-in loader with the following next.config.js:
```

**Review by ijjk — DISMISSED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 21.7s | 21.4s | -266ms |
| buildDurationCached | 4.4s | 4.5s | ⚠️ +73ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +388 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.499 | 3.515 | ⚠️ +0.02 |
| / avg req/sec | 714.46 | 711.21 | ⚠️ -3.25 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.027 | 1.869 | -0.16 |
| /error-in-render avg req/sec | 1233.1 | 1337.96 | +104.86 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 534 B | 534 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 23.2s | 24.9s | ⚠️ +1.7s |
| buildDurationCached | 4.6s | 4.9s | ⚠️ +294ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +388 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.568 | 3.729 | ⚠️ +0.16 |
| / avg req/sec | 700.65 | 670.43 | ⚠️ -30.22 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.854 | 2.075 | ⚠️ +0.22 |
| /error-in-render avg req/sec | 1348.61 | 1205.09 | ⚠️ -143.52 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 535 B | 535 B | ✓ |
| link.html gzip | 548 B | 548 B | ✓ |
| withRouter.html gzip | 529 B | 529 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: d13d1dffe568364ecf973fb4b1540069ece1a925

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d13d1dffe568364ecf973fb4b1540069ece1a925

**test/integration/invalid-document-image-import/test/index.test.js** 
- Invalid static image import in _document > Should build without error when disableStaticImages in next.config.js

<details>
<summary>Expand output</summary>

  ● Invalid static image import in _document › Should build without error when disableStaticImages in next.config.js

    command failed with code 1
    info  - Checking validity of types...
    info  - Creating an optimized production build...
    Failed to compile.

    ./public/test.jpg
    Images cannot be imported within pages/_document.js. Please move image imports that need to be displayed on every page into pages/_app.js.
    Read more: https://nextjs.org/docs/messages/custom-document-image-import
    Location: pages/_document.js

    Import trace for requested module:
    ./pages/_document.js

    ./public/test.jpg
    Module build failed: Error: Final loader (../../../packages/next/dist/build/webpack/loaders/error-loader.js) didn't return a Buffer or String

      53074 | 			if (!Buffer.isBuffer(source) && typeof source !== "string") {
      53075 | 				const currentLoader = this.getCurrentLoader(loaderContext, 0);
    > 53076 | 				const err = new Error(
            | 				            ^
      53077 | 					`Final loader (${
      53078 | 						currentLoader
      53079 | 							? compilation.runtimeTemplate.requestShortener.shorten(

      at processResult (../packages/next/dist/compiled/webpack/bundle5.js:53076:17)
      at ../packages/next/dist/compiled/webpack/bundle5.js:53162:5
      at ../packages/next/dist/compiled/webpack/bundle5.js:138606:3
      at iterateNormalLoaders (../packages/next/dist/compiled/webpack/bundle5.js:138432:10)
      at iterateNormalLoaders (../packages/next/dist/compiled/webpack/bundle5.js:138439:10)
      at ../packages/next/dist/compiled/webpack/bundle5.js:138454:3
      at runSyncOrAsync (../packages/next/dist/compiled/webpack/bundle5.js:138337:12)
      at iterateNormalLoaders (../packages/next/dist/compiled/webpack/bundle5.js:138450:2)
      at ../packages/next/dist/compiled/webpack/bundle5.js:138423:4
      at ../packages/next/dist/compiled/webpack/bundle5.js:53136:15
      Import trace for requested module:
      ./pages/_document.js
      > Build failed because of webpack errors
      at ChildProcess.<anonymous> (lib/next-test-utils.js:183:11)

</details>

**Review by timneutkens — DISMISSED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 23.3s | 23.5s | ⚠️ +220ms |
| buildDurationCached | 5.1s | 4.9s | -274ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +494 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.938 | 3.927 | -0.01 |
| / avg req/sec | 634.9 | 636.69 | +1.79 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.091 | 2.101 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 1195.85 | 1190.07 | ⚠️ -5.78 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 515 B | 515 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 26s | 26s | -5ms |
| buildDurationCached | 4.9s | 4.9s | ⚠️ +81ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +494 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.946 | 3.872 | -0.07 |
| / avg req/sec | 633.63 | 645.61 | +11.98 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.115 | 2.167 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 1181.93 | 1153.43 | ⚠️ -28.5 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 534 B | 534 B | ✓ |
| withRouter.html gzip | 516 B | 516 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
Commit: 34258bb85d7d674f8ae38d7f63c030037151e6a7

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 23.7s | 24.1s | ⚠️ +346ms |
| buildDurationCached | 5s | 5s | -16ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +496 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.99 | 3.889 | -0.1 |
| / avg req/sec | 626.53 | 642.91 | +16.38 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.194 | 2.225 | ⚠️ +0.03 |
| /error-in-render avg req/sec | 1139.56 | 1123.81 | ⚠️ -15.75 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 515 B | 515 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| buildDuration | 26.4s | 26.4s | -27ms |
| buildDurationCached | 5s | 5.1s | ⚠️ +34ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +496 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.931 | 3.966 | ⚠️ +0.04 |
| / avg req/sec | 635.94 | 630.34 | ⚠️ -5.6 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.194 | 2.148 | -0.05 |
| /error-in-render avg req/sec | 1139.65 | 1163.84 | +24.19 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix-next-disable-image-import-document | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 534 B | 534 B | ✓ |
| withRouter.html gzip | 516 B | 516 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
Commit: 628f59700132485200b5a778f96dba6d59161593

**Pull request merged**